pkg/buildkit/buildkit.go

```diff
@@ -60,6 +68,22 @@ func (b *clientBuilder) Build(ctx context.Context) (*Client, error) {
 		return nil, fmt.Errorf("failed to create buildkit client: %w", err)
 	}
 
+	var lastErr error
+
+	b.log.Info("Confirming buildkitd connectivity")
+	err = wait.ExponentialBackoffWithContext(ctx, clientCheckBackoff, func() (done bool, err error) {
+		if _, lastErr = bk.ListWorkers(ctx); lastErr != nil {
+			b.log.V(1).Info("Buildkitd is not ready")
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to contact buildkitd after %d attempts: %w", clientCheckBackoff.Steps, lastErr)
```

Comment:
shouldn't you just log the failure here and return the error instead?

Reply:
I wish. `wait.ExponentialBackoffWithContext` will stop immediately if you return an error. 😞 

https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff

Reply:
it does

https://github.com/kubernetes/apimachinery/blob/v0.24.2/pkg/util/wait/wait.go#L421

Reply:
I guess in this case I don't care about the err, maybe that's wrong. I just want the operation to succeed.

Reply:
Correct, but normally I would let the pod crashloop so alerting can take over as it prolongs. So its a balance between retrying enough times vs giving up.
If o11y was a thing, crashlooping is way easier to monitor than adding more metrics to report a prolonged issue where no progress is made.